src/hawkbit-client.c

```diff
@@ -682,7 +682,13 @@ static gboolean process_deployment(JsonNode *req_root, GError **error)
         artifact = g_new0(struct artifact, 1);
         artifact->version = json_get_string(json_chunk, "$.version", NULL);
         artifact->name = json_get_string(json_chunk, "$.name", NULL);
-        artifact->size = json_get_int(json_artifact, "$.size");
+        artifact->size = json_get_int(json_artifact, "$.size", NULL);
```

Comment:
Is there any advantage in adding this check here already? It is not noted in the commit and it checks for the return value of `json_get_int`, which not necessarily means a parsing error but could also only return the valid value `0`.

Reply:
True, I'll drop this check here. The (to be created) `hawkbit-client` refactoring PR will then contain a check for `0 && error`.

Reply:
Done.